### PR TITLE
Switch to ttlcache v3

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Update ttlcache to the latest v3 release and adapt all code paths.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go [hunk 1]
```diff
@@ -104,14 +110,10 @@ func New(logger logr.Logger) *BpfRecorder {
...
-	for _, cache := range []*ttlcache.Cache{
+	for _, cache := range []*ttlcache.Cache[string, string]{
 		b.containerIDCache, b.syscallNamesForIDCache,
 	} {
-		if err := b.SetTTL(cache, defaultCacheTimeout); err != nil {
-			return fmt.Errorf("set cache timeout: %w", err)
-		}
-		cache.SetCacheSizeLimit(maxCacheItems)
-		defer cache.Close() // nolint:gocritic // this is intentional
+		go cache.Start()
```
Comment (JAORMX) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:116]: don't we still need to set the cachesizelimit?

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go [hunk 2]
```diff
@@ -758,23 +760,16 @@ func (b *BpfRecorder) unload() {
...
-		err, ttlcache.ErrNotFound,
-	) {
-		nameString, ok := name.(string)
-		if !ok {
-			return "", errors.New("name is not a string")
-		}
-		return nameString, nil
+	item := b.syscallNamesForIDCache.Get(key)
+	if item != nil {
+		return item.Value(), nil
```
Comment (JAORMX) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:765]: this looks way cleaner!

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:116]: We do above where we create the cache :)

**Review by pjbgf — APPROVED** [review]
/lgtm

**Pull request merged**